### PR TITLE
Feature/102/login signup forms/alicat

### PR DESCRIPTION
**Pull request by adowski**
Collaborated with @cathleenjia !!!! :) <3 xD rawr

## Style and Create Login Component

- Changed login form from Rails to React
- Created `LogInPage` React component
- Styled form to align with Figma design

### Related PRs

- Added the beginnings of error handling which will be completed with Melody and Cass's PR

### Migrations

None

### Tests Performed, Edge Cases

None

### Screenshots

image

CC: @wilsonw926

### CODE REVIEW

File: app/javascript/helpers/api_routes.js [hunk 1]
```diff
@@ -18,7 +18,8 @@ class ApiRoutes {
 
   get users() {
     return {
-      delete: () => '/users/sign_out'
+      // create: () => '/users/sign_in',
```
Comment (wilsonw926) [inline app/javascript/helpers/api_routes.js]: Let's use this file!

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -2,4 +2,15 @@
 
 class SessionsController < Devise::SessionsController
   skip_before_action :require_login
+
```
Comment (wilsonw926) [inline app/controllers/sessions_controller.rb]: Delete the commented out code.

File: app/javascript/helpers/api_routes.js (hunk 1 above)
Comment (wilsonw926) [inline app/javascript/helpers/api_routes.js]: Use this file instead of hard-coding the link.

File: app/javascript/components/LogInPage.jsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+   }
+ }
+
+ handleRememberMe = (event) => {
+   this.setState({ remember_me: !this.state.remember_me });
+ }
+
+ handleSubmit = (event) => {
+   event.preventDefault();
+   const login_route = '/users/sign_in';
```
Comment (wilsonw926) [inline app/javascript/components/LogInPage.jsx]: Use the API Route here.

File: app/javascript/components/LogInPage.jsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+   const value = event.target.value;
+   if (name === "email") {
+     this.setState({ email: value });
+   } else {
+     this.setState({ password: value });
+   }
+ }
+
+ handleRememberMe = (event) => {
+   this.setState({ remember_me: !this.state.remember_me });
```
Comment (wilsonw926) [inline app/javascript/components/LogInPage.jsx]: do "1 - this.state.remember_me" here to keep the value an integer.

File: app/javascript/components/LogInPage.jsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+   event.preventDefault();
+   const login_route = '/users/sign_in';
+   let payload = {
+     email: this.state.email,
+     password: this.state.password,
+     remember_me: this.state.remember_me
+   }
+   payload = JSON.stringify({ user: payload });
+   console.log(payload)
+   debugger
```
Comment (wilsonw926) [inline app/javascript/components/LogInPage.jsx]: remove console.log and debugger

File: app/controllers/sessions_controller.rb (hunk 2 above)
Comment (adowski) [reply app/controllers/sessions_controller.rb]: done!

File: app/javascript/helpers/api_routes.js (hunk 1 above)
Comment (adowski) [reply app/javascript/helpers/api_routes.js]: added a sessions.create() function but it breaks everything ): using literal route for now, hopefully we will figure out why putting it in api_routes doesn't work

File: app/javascript/components/LogInPage.jsx (hunk 3 above)
Comment (adowski) [reply app/javascript/components/LogInPage.jsx]: see above ^^^

File: app/javascript/components/LogInPage.jsx (hunk 4 above)
Comment (adowski) [reply app/javascript/components/LogInPage.jsx]: done!

File: app/javascript/components/LogInPage.jsx (hunk 5 above)
Comment (adowski) [reply app/javascript/components/LogInPage.jsx]: done!
